Parser/ContainerCallVisitor.php

```diff
@@ -0,0 +1,117 @@
+<?php
+
+
+namespace ChameleonSystem\UpgradeHelperBundle\Parser;
+
+
+use ChameleonSystem\UpgradeHelperBundle\DataModel\CallToContainer;
+use PhpParser\Node;
+use PhpParser\Node\Name;
+use PhpParser\NodeVisitorAbstract;
+
+/**
+ * ContainerCallVisitor will visit every call to a container/service locator `get` method and produce
+ * a `CallToContainer` instance for every occurrence.
+ */
+final class ContainerCallVisitor extends NodeVisitorAbstract
+{
+    /**
+     * @var CallToContainer[]
+     */
+    private $calls;
+    /**
+     * @var string
+     */
+    private $fileName;
+    /**
+     * @var TypeResolveVisitor
+     */
+    private $typeResolver;
+
+    public function __construct(TypeResolveVisitor $typeResolver, string $fileName)
+    {
+        $this->calls = [];
+        $this->fileName = $fileName;
+        $this->typeResolver = $typeResolver;
+    }
+
+
+    public function leaveNode(Node $node)
+    {
+        if ($node instanceof Node\Expr\MethodCall) {
+            if (!property_exists($node->name, 'name')) {
+                return null;
+            }
+            if ($node->name->name !== 'get') {
+                return null;
+            }
+
+            // calling on attribute of `this`
+            if (!property_exists($node, 'var')) {
+                return null;
+            }
+            if (!property_exists($node->var, 'var')) {
+                return null;
+            }
+            if ($node->var->var->name === 'this') {
+                $alias = $node->var->name->name;
+                $resolvedClassName = $this->typeResolver->getAttributeType($alias);
+                if ($resolvedClassName !== '\Symfony\Component\DependencyInjection\Container' && $resolvedClassName !== '\Symfony\Component\DependencyInjection\ContainerInterface') {
+                    return null;
+                }
+
+                $this->calls[] = $this->createCall($node, CallToContainer::SOURCE_CONTAINER);
+            }
+
+            return null;
+        }
+
+        if ($node instanceof Node\Expr\StaticCall) {
+            if (!property_exists($node->name, 'name')) {
```

Comment:
Instead of 2 long unrelated blocks, it'd probably be a bit cleaner to move both of those blocks into methods making `leaveNode` to something like this:

```php
if ($node instanceof Node\Expr\MethodCall) {
    $this->handleMethodCall($node);
}
if ($node instanceof Node\Expr\StaticCall) {
    $this->handleStaticCall($node);
}
```

Reply:
true. Done

Reply:
:ok: